analyzers/src/SonarAnalyzer.Common/Rules/Utilities/TokenTypeAnalyzerBase.cs

```diff
@@ -39,6 +39,8 @@ public abstract class TokenTypeAnalyzerBase : UtilityAnalyzerBase<TokenTypeInfo>
 
         protected abstract TokenClassifierBase GetTokenClassifier(SyntaxToken token, SemanticModel semanticModel);
 
+        protected override bool SkipAnalysisForLargeFiles => true;
```

Comment:
This will completely disable the highlighting for the file. Even for known keywords that are cheap to detect.

We should just avoid `ClassifyIdentifier()` when the file is known to be too large.

Reply:
Indeed this would be nice but I did not want to have a different behavior per analyzer type.

Reply:
This will introduce significant decrease of UX for LTS users for next 1,5 years :( The files without highlighting are way harder to read

Reply:
one thing to consider: are we introducing a performance regression for all codebases just to avoid the edge case of huge files?